src/ui_manager.py

```diff
@@ -303,7 +303,32 @@ def stash_all_items(self, num_loot_columns: int, item_finder: ItemFinder):
                 mouse.release(button="left")
                 wait(0.4, 0.6)
                 keyboard.send("enter") #if stash already full of gold just nothing happens -> gold stays on char -> no popup window
+                wait(1.0, 1.2)
+                inventory_no_gold = self._template_finder.search("INVENTORY_NO_GOLD", self._screen.grab(), roi=self._config.ui_roi["inventory_gold"], threshold=0.97)
+                if not inventory_no_gold.valid:
+                    Logger.info("Stash tab is full of gold, selecting next stash tab.")
+                    self._curr_stash["gold"] += 1
+                    if self._curr_stash["gold"] > 3:
+                        inventory_full_gold = self._template_finder.search("INVENTORY_FULL_GOLD", self._screen.grab(), roi=self._config.ui_roi["inventory_gold"], threshold=0.98)
```

Comment:
Wouldnt this also trigger if by chance char has exactly 10000 gold? But I understand that's the best we can do cause we don't know char lvl and that would be a edge case that might not be worth the hustle.